openhands/controller/agent_controller.py

```diff
@@ -842,6 +842,8 @@ async def _step(self) -> None:
                     'contextwindowexceedederror' in error_str
                     or 'prompt is too long' in error_str
                     or 'input length and `max_tokens` exceed context limit' in error_str
+                    or 'please reduce the length of either one'
+                    in error_str  # For OpenRouter context window errors
```

Comment:
oof this is a really bad way to catch errors. Are there not more fine-grained error types we can filter for here?